crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -581,4 +582,37 @@ impl BeaconState {
         }
         Ok((rewards, penalties))
     }
+
+    pub fn process_block_header(mut self, block: BeaconBlock) -> anyhow::Result<()> {
```

Comment:
```suggestion
    pub fn process_block_header(&mut self, block: BeaconBlock) -> anyhow::Result<()> {
```

we shouldn't be consuming self